### PR TITLE
Arm death protection TELEOP

### PR DESCRIPTION
**Pull request by stardust4089**
This is code that should prevent the telescope from dying during rotation. It isn't pretty, but it should work. The individual tick values will be updated to percentages once PR #34 is merged. Includes:
- Protections in the rotator default command to retract the telescope
- Exceptions for very small adjustments
- Full allowance if telescope is retracted

Note that this does not affect autonomous setpoint movement. This is accounted for in PR #34, again.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -236,6 +236,8 @@ public static final class RotatorConstants {
     public static final double ROTATOR_SPEED_MULTIPLIER = 1.0;
 
     public static final double RotatorJoystickDeadband = 0.05;
+
+    public static final double MAX_VELOCITY_BEFORE_PULLING_IN = Integer.MAX_VALUE;
```
[the remark this answers is not in the document]
Comment (stardust4089) [reply src/main/java/frc/robot/Constants.java:240]: leftover, removed.

File: src/main/java/frc/robot/subsystems/RotatorSubsystem.java [hunk 2]
```diff
@@ -106,8 +110,6 @@ public void periodic() {
         SmartDashboard.putNumber("safe speed: " , safeSpeed);
         SmartDashboard.putNumber("Current Pos:", Math.toDegrees(getMeasurement()));
 
-        
-
 
         setSpeed(speed+ff);
```
[the remark this answers is not in the document]
Comment (stardust4089) [reply src/main/java/frc/robot/subsystems/RotatorSubsystem.java:114]: Fixed!

File: src/main/java/frc/robot/subsystems/RotatorSubsystem.java [hunk 3]
```diff
@@ -89,6 +89,10 @@ private void setSpeed(double speed) {
         rotationMotor.set(speed); // Defaults to PercentOutput
     }
 
+    private double getVelocity(){
```
[the remark this answers is not in the document]
Comment (stardust4089) [reply src/main/java/frc/robot/subsystems/RotatorSubsystem.java:92]: Correct, removed

**Review by mikemag — APPROVED** [review]
Cool

**Review by oliverstivers — APPROVED** [review]
looks good, looking forward to the arm not dying

**Pull request merged**